pkgs/top-level/all-packages.nix

```diff
@@ -14104,23 +14104,17 @@ let
 
   cuyo = callPackage ../games/cuyo { };
 
-  dfhack = callPackage_i686 ../games/dfhack {
-    inherit (pkgsi686Linux.perlPackages) XMLLibXML XMLLibXSLT;
-  };
-
   dhewm3 = callPackage ../games/dhewm3 {};
 
   drumkv1 = callPackage ../applications/audio/drumkv1 { };
 
-  dwarf_fortress = callPackage_i686 ../games/dwarf-fortress {
-    SDL_image = pkgsi686Linux.SDL_image.override {
-      libpng = pkgsi686Linux.libpng12;
-    };
-    inherit (pkgsi686Linux.perlPackages) XMLLibXML XMLLibXSLT;
+  dwarf-fortress-packages = callPackage ../games/dwarf-fortress { };
+
+  dwarf-fortress = dwarf-fortress-packages.dwarf-fortress.override {
     enableDFHack = config.dwarfFortress.enableDFHack or false;
```

Comment:
Oh, I completely forgot about that part.

AFAIK we have a policy of avoiding `config` options in favor of package
overrides, especially when an option doesn't affect multiple packages
(like `pulseaudio`). However, we already have one for DFHack. So we have
three choices:

1. Remove existing option, breaking some setups but consistently getting
rid of it;
2. Leave it as-is as a message that we don't support config options more
and hope to deal with this later;
3. Add a new option, getting consistency and compatibility but breaking
the policy even more.

I think now is a good moment to make choices so let's avoid (2). This
leaves us with (1) vs (3). Personally I'm a little in favor of dropping
options -- we are renaming the package as well, breaking compatibility
already. What's your opinion on this?

On January 24, 2016 8:14:47 AM GMT+03:00, Adam Russell
<notifications@github.com> wrote:

    In pkgs/top-level/all-packages.nix
    <https://github.com/NixOS/nixpkgs/pull/12339#discussion_r50630354>:

    >    dhewm3 = callPackage ../games/dhewm3 {};
    >  
    >    drumkv1 = callPackage ../applications/audio/drumkv1 { };
    >  
    > -  dwarf_fortress = callPackage_i686 ../games/dwarf-fortress {
    > -    SDL_image = pkgsi686Linux.SDL_image.override {
    > -      libpng = pkgsi686Linux.libpng12;
    > -    };
    > -    inherit (pkgsi686Linux.perlPackages) XMLLibXML XMLLibXSLT;
    > +  dwarf-fortress-packages = callPackage ../games/dwarf-fortress { };
    > +
    > +  dwarf-fortress = dwarf-fortress-packages.dwarf-fortress.override {
    >      enableDFHack = config.dwarfFortress.enableDFHack or false;

    Also add |theme = config.dwarfFortress.theme or null;|

    —
    Reply to this email directly or view it on GitHub
    <https://github.com/NixOS/nixpkgs/pull/12339/files#r50630354>.